src/node_shadow_realm.cc

```diff
@@ -137,6 +149,25 @@ v8::MaybeLocal<v8::Value> ShadowRealm::BootstrapRealm() {
     }
   }
 
+  if (ExecuteBootstrapper(
+          "internal/bootstrap/switches/does_not_own_process_state")
```

Comment:
What's the idea about choosing which script to run and which not to run? This just sets up a bunch of states in process, but there are also a bunch of other states initialized in `bootstrap/node.js` too. I wonder if we should move those to `realm.js` and just leave the `globalThis` mutations there..

Reply:
This is needed to set up the process properties for built-in modules exclusively since `process` is not exposed to ShadowRealm yet. I added a code comment to explain that `process.cwd()` initialized in this switch is needed by the ESM loader.

I believe a bunch of global properties like `Buffer` can be moved to `realm.js` indeed. I can do it in a follow-up work.

Reply:
I am okay with a follow-up though want to clarify that I was talking about the "adding things to `process`"  part in `node.js`, not the "exposing `process` to `globalThis` part - the former could just be moved to realm.js, if we are already loading `does_not_own_process_state` here which is also just "adding things to `process`".